nano/node/node.cpp

```diff
@@ -1085,19 +1111,57 @@ class distributed_work : public std::enable_shared_from_this<distributed_work>
 					boost::property_tree::write_json (ostream, request);
 					request_string = ostream.str ();
 				}
-				boost::beast::http::request<boost::beast::http::string_body> request;
-				request.method (boost::beast::http::verb::post);
-				request.set (boost::beast::http::field::content_type, "application/json");
-				request.target ("/");
-				request.version (11);
-				request.body () = request_string;
-				request.prepare_payload ();
-				auto socket (std::make_shared<boost::asio::ip::tcp::socket> (this_l->node->io_ctx));
-				boost::beast::http::async_write (*socket, request, [socket](boost::system::error_code const & ec, size_t bytes_transferred) {
+				auto request (std::make_shared<boost::beast::http::request<boost::beast::http::string_body>> ());
+				request->method (boost::beast::http::verb::post);
+				request->set (boost::beast::http::field::content_type, "application/json");
+				request->target ("/");
+				request->version (11);
+				request->body () = request_string;
+				request->prepare_payload ();
+
+				boost::beast::http::async_write (cancelling->socket, *request, [this_l, request, cancelling](boost::system::error_code const & ec, size_t bytes_transferred) {
+					if (ec)
+					{
+						this_l->node->logger.try_log (boost::str (boost::format ("Unable to send work_cancel to work_peer %1% %2%: %3% (%4%)") % cancelling->address % cancelling->port % ec.message () % ec.value ()));
+					}
 				});
-			});
+			}
+		});
+	}
+	void stop (bool const local_stop)
+	{
+		std::lock_guard<std::mutex> lock (mutex);
+		if (!stopped)
```

Comment:
Perhaps use `if (!stopped.exchange (true))` like in node::stopped - can also move the lock inside the scope as it's an atomic.